extension-manifest-v2/app/panel/reducers/blocking.js

```diff
@@ -69,6 +70,31 @@ function mergeTrackers(adblockerTrackers, antiTrackingTrackers) {
 			all.set(tracker.name, existing);
 		}
 	});
+	const findTrackerByWhitelistedDomain = domain => Array.from(all.values()).find(
+		tracker => !!tracker.domains.some(whitelistedDomain => whitelistedDomain.endsWith(domain))
+	);
+
+	const pageTld = parse(pageUrl).domain;
+
+	Object.keys(whitelist).forEach((domain) => {
+		if (!whitelist[domain].hosts.some(host => host === pageTld)) {
+			return;
+		}
+		const tld = parse(domain).domain;
+		const existing = findTrackerByWhitelistedDomain(tld) || all.get(tld);
```

Comment:
```suggestion
		const existing = all.get(tld) || findTrackerByWhitelistedDomain(tld);
```

should be equivalent, right? Unless it is extremely unlikely that `all.get(tld)` will find something, then trying the single map look first (before going into the loops) would be good (people could configure a huge whitelist).